Description: do not pass -Wl,--gc-sections on powerpc, s390, arm* and mips* when building
             libQtWebKit.so; works around a binutils bug that results in a
             segfault on powerpc, internal link error on s390 and OOM on armel/mipsel.
Author: Steve Langasek <vorlon@debian.org>
Author: Modestas Vainius <modax@debian.org>
Origin: vendor
Forwarded: no
Last-Update: 2011-03-28 (mips* and arm* added)

---
 Source/WebCore/WebCore.pri |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Source/WebCore/WebCore.pri
+++ b/Source/WebCore/WebCore.pri
@@ -326,7 +326,7 @@ mac {
 }
 
 unix:!mac:*-g++*:QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections
-unix:!mac:*-g++*:QMAKE_LFLAGS += -Wl,--gc-sections
+unix:!mac:*-g++*:!equals(QT_ARCH, powerpc):!equals(QT_ARCH, s390):!equals(QT_ARCH, mips):!equals(QT_ARCH, arm):QMAKE_LFLAGS += -Wl,--gc-sections
 linux*-g++*:QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 unix|win32-g++*:QMAKE_PKGCONFIG_REQUIRES = QtCore QtGui QtNetwork
